spec/controllers/catalog_controller_spec.rb

```diff
@@ -117,4 +117,75 @@
       end
     end
   end
+
+  describe "switching search algorithms" do
+    let(:repository) { CatalogController.blacklight_config.repository }
+
+    before do
+      allow(repository).to receive(:search).and_call_original
+      allow_any_instance_of(Blacklight::Configuration).to receive(:repository).and_return(repository)
```

Comment:
I wonder if we can utilize the `controller` variable here.  I did not think of it when I was playing with the tests
```suggestion
      allow(controller.blacklight_config).to receive(:repository).and_return(repository)
```